gnd/src/main/java/com/google/android/gnd/ui/home/HomeScreenViewModel.java

```diff
@@ -110,6 +115,25 @@
                         .onErrorReturnItem(false)));
   }
 
+  /** Handle state of the UI elements depending upon the active project. */
+  private void onActivateProject(Loadable<Project> project) {
+    Boolean addBtnVisibility =
+        Optional.of(project)
+            .filter(Loadable::isLoaded)
+            .map(p -> containsLayers(p.value()))
+            .orElse(false);
+    addFeatureBtnVisibility.postValue(addBtnVisibility ? View.VISIBLE : View.GONE);
+  }
+
+  /** Returns true if the given project contains layers, otherwise false. */
+  private boolean containsLayers(Optional<Project> project) {
+    return project.map(p -> !p.getLayers().isEmpty()).orElse(false);
```

Comment:
I think you can drop `orElse` here and call `flatMap` on line 123.

Reply:
Done